src/main/java/org/jenkinsci/plugins/workflow/actions/ArgumentsAction.java

```diff
@@ -0,0 +1,169 @@
+ * Supplied argument values can be hidden and replaced with a {@link NotStoredReason} for security or performance.
+ */
+public abstract class ArgumentsAction implements PersistentAction {
+
+    /** Used as a placeholder for arguments not stored for various reasons */
+    public enum NotStoredReason {
+        /** Denotes an unsafe value that cannot be stored/displayed due to sensitive info */
+        MASKED_VALUE,
+
+        /** Denotes an object that is too big to retain, such as strings exceeding {@link #MAX_STRING_LENGTH} */
+        OVERSIZE_VALUE
+    }
+
+    /** Largest string we'll persist in the Step, for performance reasons */
+    public static final int MAX_STRING_LENGTH = 1024;
+
+    @Override
+    public String getIconFileName() {
+        return null;
+    }
+
+    @Override
+    public String getDisplayName() {
+        return "Step Arguments";
+    }
+
+    @Override
+    public String getUrlName() {
+        return "stepArguments";
+    }
+
+    /**
+     * Get the map of arguments for the {@link Step}, with a {@link NotStoredReason} instead of the value
+     *  if part of the arguments are not retained.
+     * @return The arguments for the Step.
+     */
+    @Nonnull
+    public Map<String,Object> getArguments() {
+        return Collections.unmodifiableMap(getArgumentsInternal());
+    }
+
+    @Nonnull
+    public static Map<String,Object> getArguments(@Nonnull FlowNode n) {
+        ArgumentsAction aa = n.getPersistentAction(ArgumentsAction.class);
+        return aa != null ? aa.getArguments() : Collections.EMPTY_MAP;
+    }
+
+    /**
+     * Get just the fully stored, non-null arguments
+     * This means the arguments with all {@link NotStoredReason} or null values removed
+     * @return Map of all completely stored arguments
+     */
+    @Nonnull
+    public Map<String, Object> getFilteredArguments() {
+        Map<String, Object> internalArgs = this.getArgumentsInternal();
+        if (internalArgs.size() == 0) {
+            return Collections.emptyMap();
+        }
+        HashMap<String, Object> filteredArguments = new HashMap<String, Object>();
+        for (Map.Entry<String, Object> entry : internalArgs.entrySet()) {
+            if (entry.getValue() != null && !(entry.getValue() instanceof NotStoredReason)) {
+                filteredArguments.put(entry.getKey(), entry.getValue());
+            }
+        }
+        return filteredArguments;
+    }
+
+    /**
+     * Get just the fully stored, non-null arguments
+     * This means the arguments with all {@link NotStoredReason} or null values removed
+     * @param n FlowNode to get arguments for
+     * @return Map of all completely stored arguments
+     */
+    @Nonnull
+    public static Map<String, Object> getFilteredArguments(@Nonnull FlowNode n) {
+        ArgumentsAction act = n.getPersistentAction(ArgumentsAction.class);
+        return act != null ? act.getFilteredArguments() : Collections.EMPTY_MAP;
```

Comment:
Well, depending on the Java source level you may need to use

```java
Collections.<Map<String, Object>>emptyMap()
```

IIRC in a newer level the inference is improved.